File: 977.js
function sortedSquares(arr) {
    let l = 0
    let r = arr.length - 1
    let p = r
    let ans = []
    while (l <= r) {
        if (arr[r] < Math.abs(arr[l])) ans[p--] = arr[l++] ** 2
        else ans[p--] = arr[r--] ** 2
    }
    return ans
}

console.log(sortedSquares([-4, -1, 0, 3, 10]));

File: Algo1/977.py
class Solution:
    def sortedSquares(self, arr: List[int]) -> List[int]:
        ans = [0] * len(arr)

        l, r = 0, len(arr) - 1
        p = r
        while l <= r:
            if arr[l] ** 2 < arr[r] ** 2:
                ans[p] = arr[r] ** 2
                r = r - 1
            else:
                ans[p] = arr[l] ** 2
                l = l + 1
            p = p - 1
        return ans
